### PR TITLE
Add new transaction methods

### PR DESCRIPTION
**Pull request by qwertyniop1**

### CODE REVIEW

File: spec/client_spec.rb [hunk 1]
```diff
@@ -149,4 +149,175 @@
...
+        expect(response.payment_method).to be_empty
+        expect(response.id).to be_nil
+        expect(response.transaction).to be_nil
+        expect(response.errors).to eq ({ "request" => { "amount" => ["is missing"] } })
+      end
+    end
+
+    describe '#payment' do
+      before do
+        stub_request(:post, /api.bepaid.by\/beyag\/transactions\/payment/).to_return(response_obj)
```
Comment (avsh) [inline spec/client_spec.rb]: api.begateway.com

File: spec/client_spec.rb [hunk 2]
```diff
@@ -149,4 +149,175 @@
...
+      context 'failed request' do
+        let(:response_obj) { { body: failed_response.to_json, status: 422 } }
+
+        it_behaves_like 'failed beyag response', :payment
+      end
+    end
+
+    describe '#refund' do
+      before do
+        stub_request(:post, /api.bepaid.by\/beyag\/transactions\/refund/).to_return(response_obj)
```
Comment (avsh) [inline spec/client_spec.rb]: api.begateway.com

File: spec/client_spec.rb [hunk 3]
```diff
@@ -149,4 +149,175 @@
...
+      context 'failed request' do
+        let(:response_obj) { { body: failed_response.to_json, status: 422 } }
+
+        it_behaves_like 'failed beyag response', :refund
+      end
+    end
+
+    describe '#payout' do
+      before do
+        stub_request(:post, /api.bepaid.by\/beyag\/transactions\/payout/).to_return(response_obj)
```
Comment (avsh) [inline spec/client_spec.rb]: api.begateway.com

**Comment by qwertyniop1** [comment]
:up:

**Pull request merged**